File: examples/c/fizzbuzz.c
#include <assert.h>
#include <stdio.h>
#include <string.h>

static const char *fizzbuzz(int number) {
    if (number % 15 == 0) {
        return "FizzBuzz";
    }
    if (number % 3 == 0) {
        return "Fizz";
    }
    if (number % 5 == 0) {
        return "Buzz";
    }
    return "number";
}

int main(void) {
    assert(strcmp(fizzbuzz(15), "FizzBuzz") == 0);
    assert(strcmp(fizzbuzz(3), "Fizz") == 0);
    assert(strcmp(fizzbuzz(5), "Buzz") == 0);
    assert(strcmp(fizzbuzz(7), "number") == 0);
    printf("fizzbuzz: ok\n");
    return 0;
}

File: examples/java/FizzBuzz.java
public class FizzBuzz {
    static String labelFor(int number) {
        if (number % 15 == 0) {
            return "FizzBuzz";
        }
        if (number % 3 == 0) {
            return "Fizz";
        }
        if (number % 5 == 0) {
            return "Buzz";
        }
        return "number";
    }

    static void check(boolean condition, String message) {
        if (!condition) {
            throw new AssertionError(message);
        }
    }

    public static void main(String[] args) {
        check(labelFor(15).equals("FizzBuzz"), "15 should produce FizzBuzz");
        check(labelFor(3).equals("Fizz"), "3 should produce Fizz");
        check(labelFor(5).equals("Buzz"), "5 should produce Buzz");
        check(labelFor(7).equals("number"), "7 should produce number");
        System.out.println("fizzBuzz: ok");
    }
}
